File: apps/push-service/src/main/java/com/example/chat/push/application/PushMessageService.java
package com.example.chat.push.application;

import com.example.chat.push.domain.PushMessage;
import com.example.chat.push.domain.PushMessageRepository;
import com.example.chat.push.interfaces.kafka.NotificationEvent;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * 푸시 메시지 저장 서비스
 *
 * 책임: Kafka 이벤트를 받아 PushMessage 도메인 객체를 저장합니다.
 */
@Service
@RequiredArgsConstructor
public class PushMessageService {
    private final PushMessageRepository pushMessageRepository;

    @Transactional
    public PushMessage savePushMessage(NotificationEvent event) {
        PushMessage pushMessage = PushMessage.of(
                event.targetUserId(),
                event.title(),
                event.content(),
                event.pushType());
        return pushMessageRepository.save(pushMessage);
    }
}

File: apps/push-service/src/main/kotlin/com/example/chat/push/application/PushMessageService.kt
package com.example.chat.push.application

import com.example.chat.push.domain.PushMessage
import com.example.chat.push.domain.PushMessageRepository
import com.example.chat.push.interfaces.kafka.NotificationEvent
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class PushMessageService(private val pushMessageRepository: PushMessageRepository) {
    @Transactional
    fun savePushMessage(event: NotificationEvent): PushMessage {
        val pushMessage =
                PushMessage(
                        targetUserId = event.targetUserId,
                        title = event.title,
                        content = event.content,
                        pushType = event.pushType
                )
        return pushMessageRepository.save(pushMessage)
    }
}
